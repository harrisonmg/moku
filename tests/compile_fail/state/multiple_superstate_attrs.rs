use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    #[derive(Debug, Clone, Copy)]
    enum BlinkyState {}
    impl StateEnum for BlinkyState {}

    struct Top {}
    impl TopState<BlinkyState> for Top {}

    #[superstate(Top)]
    #[superstate(Top)]
    impl State<BlinkyState> for Bottom {
        type Superstates<'a> = std::marker::PhantomData<&'a ()>;

        fn enter<'a>(_superstates: &mut Self::Superstates<'a>) -> StateEntry<Self, BlinkyState> {
            StateEntry::State(Self {})
        }
    }
}

fn main() {}
