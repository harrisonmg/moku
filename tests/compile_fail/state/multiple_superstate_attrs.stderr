error: multiple `moku::superstate` attributes defined for this `moku::State`
  --> tests/compile_fail/state/multiple_superstate_attrs.rs:17:5
   |
17 |     #[superstate(Top)]
   |     ^
